tests/mock_crust/mod.rs

```diff
@@ -54,6 +75,51 @@ fn disconnect_on_rebootstrap() {
     expect_next_event!(unwrap!(nodes.last_mut()), Event::Terminated);
 }
 
+#[test]
+fn candidate_timeout_resource_proof() {
+    let network = Network::new(MIN_SECTION_SIZE, None);
+    let mut nodes = create_connected_nodes_until_split(&network, vec![1, 1], false);
+    let bootstrap_config = BootstrapConfig::with_contacts(&[nodes[0].handle.endpoint()]);
+    nodes.insert(
+        0,
+        TestNode::builder(&network)
+            .bootstrap_config(bootstrap_config)
+            .create(),
+    );
+
+    // Initiate connection until the node receive RelocateResponse:
+    info!("Candidate joining name: {}", nodes[0].name());
+    poll_and_resend_until(&mut nodes, &mut [], &|nodes| {
+        nodes[0].inner.is_proving_node()
+    });
+    let proving_node = nodes.remove(0);
+
+    assert!(
+        proving_node.inner.is_proving_node(),
+        "Accepted as candidate"
+    );
+
+    // Continue without the joining node until all node consensused events:
```

Comment:
> node consensused events:
maybe switch to `nodes idle` poll_and_resend isnt just abt consensus events, also includes ack_timeouts and stuff ofc.